2007-03-25  Thomas Schwinge  <tschwinge@gnu.org>

	* isofs/inode.c (read_disknode): Correctly convert from nanoseconds to
	microseconds.

2007-03-30  Thomas Schwinge  <tschwinge@gnu.org>

	* inode.c (read_node): Correctly convert from nanoseconds to
	microseconds.

===================================================================
RCS file: /sources/hurd/hurd/fatfs/inode.c,v
retrieving revision 1.5
retrieving revision 1.6
diff -u -r1.5 -r1.6
--- hurd/fatfs/inode.c	2003/08/06 21:47:57	1.5
+++ hurd/fatfs/inode.c	2007/03/30 14:15:49	1.6
@@ -384,7 +387,7 @@
     fat_to_epoch (dr->write_date, dr->write_time, &ts);
     st->st_ctime = st->st_mtime = st->st_atime = ts.tv_sec;
     st->st_ctime_usec = st->st_mtime_usec = st->st_atime_usec
-      = ts.tv_nsec * 1000;
+      = ts.tv_nsec / 1000;
   }
   
   st->st_blksize = bytes_per_sector;
===================================================================
RCS file: /sources/hurd/hurd/isofs/inode.c,v
retrieving revision 1.16
retrieving revision 1.17
diff -u -r1.16 -r1.17
--- hurd/isofs/inode.c	2002/08/06 12:33:17	1.16
+++ hurd/isofs/inode.c	2007/03/25 20:29:33	1.17
@@ -1,5 +1,5 @@
 /*
-   Copyright (C) 1997,98,2002 Free Software Foundation, Inc.
+   Copyright (C) 1997, 1998, 2002, 2007 Free Software Foundation, Inc.
    Written by Thomas Bushnell, n/BSG.
 
    This file is part of the GNU Hurd.
@@ -433,7 +433,7 @@
       isodate_915 (dr->date, &ts);
       st->st_ctime = st->st_mtime = st->st_atime = ts.tv_sec;
       st->st_ctime_usec = st->st_mtime_usec = st->st_atime_usec
-	= ts.tv_nsec * 1000;
+	= ts.tv_nsec / 1000;
     }
 
   /* Override what we have better info for */
@@ -442,19 +442,19 @@
       if (rl->tfflags & TF_CREATION)
 	{
 	  st->st_ctime = rl->ctime.tv_sec;
-	  st->st_ctime_usec = rl->ctime.tv_nsec * 1000;
+	  st->st_ctime_usec = rl->ctime.tv_nsec / 1000;
 	}
 
       if (rl->tfflags & TF_ACCESS)
 	{
 	  st->st_atime = rl->atime.tv_sec;
-	  st->st_atime_usec = rl->atime.tv_nsec * 1000;
+	  st->st_atime_usec = rl->atime.tv_nsec / 1000;
 	}
 
       if (rl->tfflags & TF_MODIFY)
 	{
 	  st->st_mtime = rl->mtime.tv_sec;
-	  st->st_mtime_usec = rl->mtime.tv_nsec * 1000;
+	  st->st_mtime_usec = rl->mtime.tv_nsec / 1000;
 	}
     }
 
